modules/ml/src/main/java/org/apache/ignite/ml/naivebayes/discrete/DiscreteNaiveBayesTrainer.java

```diff
@@ -0,0 +1,238 @@
+     * @param lbExtractor Label extractor.
+     * @return Model.
+     */
+    @Override public <K, V> DiscreteNaiveBayesModel fit(DatasetBuilder<K, V> datasetBuilder,
+        IgniteBiFunction<K, V, Vector> featureExtractor, IgniteBiFunction<K, V, Double> lbExtractor) {
+        return updateModel(null, datasetBuilder, featureExtractor, lbExtractor);
+    }
+
+    /** {@inheritDoc} */
+    @Override protected boolean checkState(DiscreteNaiveBayesModel mdl) {
+
+        if (mdl.getBucketThresholds().length != bucketThresholds.length) {
+            return false;
+        }
+        for (int i = 0; i < bucketThresholds.length; i++) {
+            for (int j = 0; i < bucketThresholds[i].length; i++) {
+                if (Math.abs(mdl.getBucketThresholds()[i][j] - bucketThresholds[i][j]) > PRECISION) {
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+
+    /** {@inheritDoc} */
+    @Override protected <K, V> DiscreteNaiveBayesModel updateModel(DiscreteNaiveBayesModel mdl,
+        DatasetBuilder<K, V> datasetBuilder, IgniteBiFunction<K, V, Vector> featureExtractor,
+        IgniteBiFunction<K, V, Double> lbExtractor) {
+
+        try (Dataset<EmptyContext, DiscreteNaiveBayesSumsHolder> dataset = datasetBuilder.build(
+            (upstream, upstreamSize) -> new EmptyContext(),
+            (upstream, upstreamSize, ctx) -> {
+                DiscreteNaiveBayesSumsHolder res = new DiscreteNaiveBayesSumsHolder();
+                while (upstream.hasNext()) {
+                    UpstreamEntry<K, V> entity = upstream.next();
+
+                    Vector features = featureExtractor.apply(entity.getKey(), entity.getValue());
+                    Double label = lbExtractor.apply(entity.getKey(), entity.getValue());
+
+                    long[][] valuesInBucket;
+
+                    int size = features.size();
+                    if (!res.valuesInBucketPerLbl.containsKey(label)) {
+                        valuesInBucket = new long[size][];
+                        for (int i = 0; i < size; i++) {
+                            valuesInBucket[i] = new long[bucketThresholds[i].length + 1];
+                            Arrays.fill(valuesInBucket[i], 0L);
+                        }
+                        res.valuesInBucketPerLbl.put(label, valuesInBucket);
+                    }
+                    if (!res.featureCountersPerLbl.containsKey(label)) {
+                        res.featureCountersPerLbl.put(label, 0);
+                    }
+                    res.featureCountersPerLbl.put(label, res.featureCountersPerLbl.get(label) + 1);
+
+                    valuesInBucket = res.valuesInBucketPerLbl.get(label);
+                    for (int j = 0; j < size; j++) {
+                        double x = features.get(j);
+                        int bucketNumber = toBucketNumber(x, bucketThresholds[j]);
+                        ++valuesInBucket[j][bucketNumber];
```

Comment:
let's use "valuesInBucket[j][bucketNumber]+=1" or "valuesInBucket[j][bucketNumber]=valuesInBucket[j][bucketNumber] + 1" style? IMHO such style more readable

Reply:
I fixed it with `+=1` style.